tests/root_gauge_factory_proxy/test_root_gauge_factory_proxy_admin.py

```diff
@@ -0,0 +1,275 @@
+
+    assert root_gauge_factory.owner() == alice
+    manager = root_gauge_factory_proxy.manager()
+    # proxy cannot set bridger yet:
+    for admin in [manager, default_ownership_admin]:
+        with brownie.reverts():
+            root_gauge_factory_proxy.set_bridger(
+                root_gauge_factory, chain.id, ETH_ADDRESS, {"from": admin}
+            )
+
+    # so transfer root gauge factory ownership to proxy:
+    root_gauge_factory.commit_transfer_ownership(root_gauge_factory_proxy, {"from": alice})
+    root_gauge_factory_proxy.accept_transfer_ownership(root_gauge_factory, {"from": bob})
+
+    for admin in [manager, default_ownership_admin]:
+        root_gauge_factory_proxy.set_bridger(
+            root_gauge_factory, chain.id, ETH_ADDRESS, {"from": admin}
+        )
+        assert root_gauge_factory.get_bridger(chain.id) == ETH_ADDRESS
+        chain.undo()
+
+    # emergency admin cannot set bridger:
+    with brownie.reverts():
+        root_gauge_factory_proxy.set_bridger(
+            root_gauge_factory, chain.id, ETH_ADDRESS, {"from": default_emergency_admin}
+        )
+
+    # but emergency admin can set itself (or anyone) as manager and change bridger:
+    root_gauge_factory_proxy.set_manager(default_emergency_admin, {"from": default_emergency_admin})
+    root_gauge_factory_proxy.set_bridger(
+        root_gauge_factory, chain.id, ETH_ADDRESS, {"from": default_emergency_admin}
+    )
+    assert root_gauge_factory.get_bridger(chain.id) == ETH_ADDRESS
+
+
+def test_set_implementation(root_gauge_factory, root_gauge_factory_proxy, alice, bob, chain):
+
+    default_ownership_admin = root_gauge_factory_proxy.ownership_admin()
+    default_emergency_admin = root_gauge_factory_proxy.emergency_admin()
+
+    assert root_gauge_factory.owner() == alice
+    manager = root_gauge_factory_proxy.manager()
+    # proxy cannot set gauge implementation contract yet:
+    for admin in [manager, default_ownership_admin]:
+        with brownie.reverts():
+            root_gauge_factory_proxy.set_implementation(
+                root_gauge_factory, ETH_ADDRESS, {"from": admin}
+            )
+
+    # so transfer root gauge factory ownership to proxy:
+    root_gauge_factory.commit_transfer_ownership(root_gauge_factory_proxy, {"from": alice})
+    root_gauge_factory_proxy.accept_transfer_ownership(root_gauge_factory, {"from": bob})
+
+    for admin in [manager, default_ownership_admin]:
+        root_gauge_factory_proxy.set_implementation(
+            root_gauge_factory, ETH_ADDRESS, {"from": admin}
+        )
+        assert root_gauge_factory.get_implementation() == ETH_ADDRESS
+        chain.undo()
+
+    # emergency admin cannot set gauge implementation:
+    with brownie.reverts():
+        root_gauge_factory_proxy.set_implementation(
+            root_gauge_factory, ETH_ADDRESS, {"from": default_emergency_admin}
+        )
+
+    # but emergency admin can set itself (or anyone) as manager and change implementation:
```

Comment:
This was already tested in `set_manager` case (i.e. that the emg admin can call set_manager), therefore adds fluff to the test case.

General ideas is to just verify that privileged accounts can call, and unprivileged can't.